### PR TITLE
ROX-25948: Generic certificate requester

### PR DESCRIPTION
**Pull request by vladbologa**
### Description

This PR uses generics instead of code duplication for `localscanner.CertificateRequester` and `securedcluster.CertificateRequester`.

Even though the Local Scanner certificate refresh mechanism will be deprecated starting with ACS 4.7, it will still be supported for backward compatibility with older Centrals that do not support the Secured Cluster cert refresh API. Since they'll have to co-exist for a longer time, it might be better if they reuse code instead of duplicating it.

### User-facing documentation

- [x] CHANGELOG is updated **OR** update is not needed
- [x] [documentation PR](https://spaces.redhat.com/display/StackRox/Submitting+a+User+Documentation+Pull+Request) is created and is linked above **OR** is not needed

### Testing and quality

- [x] the change is production ready: the change is GA or otherwise the functionality is gated by a feature flag
- [x] CI results are inspected

#### Automated testing

- [ ] added unit tests
- [ ] added e2e tests
- [ ] added regression tests
- [ ] added compatibility tests
- [x] modified existing tests

#### How I validated my change

This PR is a refactor and should not change functionality, existing tests should verify that.

I also verified with a local deployment that a set of Secured Cluster certificates is being created on Sensor startup, and that a refresh is being scheduled.

### CODE REVIEW

File: sensor/kubernetes/certrefresh/certificates/requester.go [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+		*central.IssueSecuredClusterCertsResponse,
+	](
+		sendC,
+		receiveC,
+		&SecuredClusterMessageFactory{},
+		&SecuredClusterResponseFactory{},
+	)
+}
+
+func newRequester[ReqT any, ResT ProtobufResponse](
```
Comment (porridge) [inline sensor/kubernetes/certrefresh/certificates/requester.go:61]: One optional nitpick, I cannot help but think about REST when I see `ResT`. Perhaps consider this change?
```suggestion
func newRequester[ReqT any, RespT ProtobufResponse](
```

**Review by porridge — APPROVED** [review]
I think this is actually more readable than I would expect!

If we really have to keep both APIs around for 1+ year, then I think I'd prefer this incarnation, even though it's just ~130 lines shorter than the identical-siblings version.

File: sensor/kubernetes/certrefresh/certificates/requester.go [hunk 2]
```diff
@@ -53,61 +57,68 @@ func NewSecuredClusterCertificateRequester(sendC chan<- *message.ExpiringMessage
...
-	messageFactory  MessageFactory
-	responseFactory ResponseFactory[ResT]
+type genericRequester[ReqT any, RespT protobufResponse] struct {
+	sendC             chan<- *message.ExpiringMessage
+	receiveC          <-chan RespT
+	stopC             concurrency.ErrorSignal
+	requests          sync.Map
+	messageFactory    messageFactory
+	responseFactory   responseFactory[RespT]
+	centralCapability *centralsensor.CentralCapability
```
Comment (porridge) [inline sensor/kubernetes/certrefresh/certificates/requester.go:92]: Maybe call this
```suggestion
	requiredCentralCapability *centralsensor.CentralCapability
```
?

**Review by porridge — APPROVED** [review]

File: sensor/kubernetes/certrefresh/certificates/requester.go (hunk 2 above)
Comment (vladbologa) [reply sensor/kubernetes/certrefresh/certificates/requester.go:92]: That's actually how I wanted to name it originally, will do.

**Review by jschnath — APPROVED** [review]
Nice test coverage, looks good to me

**Pull request merged**